app/internal/page/payment_confirmation_test.go

```diff
@@ -0,0 +1,232 @@
+		w := httptest.NewRecorder()
+		r, _ := http.NewRequest(http.MethodGet, "/payment-confirmation", nil)
+
+		template := &mockTemplate{}
+		dataStore := (&mockDataStore{}).
+			withLpaDataInStore()
+
+		sessionsStore := &mockSessionsStore{}
+		sessionsStore.
+			On("Get", r, "pay").
+			Return(&sessions.Session{}, expectedError)
+
+		logger := &mockLogger{}
+		logger.
+			On("Print", fmt.Sprintf("unable to retrieve session using key '%s': %s", "pay", expectedError.Error())).
+			Return(nil)
+
+		err := PaymentConfirmation(logger, template.Func, &mockPayClient{}, dataStore, sessionsStore)(appData, w, r)
+		resp := w.Result()
+
+		assert.Equal(t, expectedError, err)
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+		mock.AssertExpectationsForObjects(t, dataStore, sessionsStore, logger)
+	})
+
+	t.Run("Returns an error if unable to get payment info from payClient", func(t *testing.T) {
+		w := httptest.NewRecorder()
+		r, _ := http.NewRequest(http.MethodGet, "/payment-confirmation", nil)
+
+		dataStore := (&mockDataStore{}).
+			withLpaDataInStore()
+
+		sessionsStore := (&mockSessionsStore{}).
+			withPaySession(r)
+
+		logger := &mockLogger{}
+		logger.
+			On("Print", fmt.Sprintf("unable to retrieve payment info: %s", expectedError.Error())).
+			Return(nil)
+
+		payClient := &mockPayClient{}
+		payClient.
+			On("GetPayment", "abc123").
+			Return(pay.GetPaymentResponse{}, expectedError)
+
+		template := &mockTemplate{}
+
+		err := PaymentConfirmation(logger, template.Func, payClient, dataStore, sessionsStore)(appData, w, r)
+		resp := w.Result()
+
+		assert.Equal(t, expectedError, err)
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+		mock.AssertExpectationsForObjects(t, dataStore, sessionsStore, logger, payClient)
+	})
+
+	t.Run("Logs an error if unable to expire cookie in sessionStore", func(t *testing.T) {
+		w := httptest.NewRecorder()
+		r, _ := http.NewRequest(http.MethodGet, "/payment-confirmation", nil)
+
+		dataStore := (&mockDataStore{}).
+			withLpaDataInStore().
+			withUpdatedLpaData("abc123", "123456789012")
+
+		sessionsStore := (&mockSessionsStore{}).
+			withPaySession(r)
+
+		sessionsStore.
+			On("Save", r, w, mock.Anything).
+			Return(expectedError)
+
+		logger := &mockLogger{}
+		logger.
+			On("Print", fmt.Sprintf("unable to expire cookie in session: %s", expectedError.Error())).
+			Return(nil)
+
+		payClient := (&mockPayClient{}).
+			withASuccessfulPayment("abc123", "123456789012")
+
+		template := &mockTemplate{}
+		template.
+			On("Func", w, &paymentConfirmationData{App: appData, PaymentReference: "123456789012"}).
+			Return(nil)
+
+		err := PaymentConfirmation(logger, template.Func, payClient, dataStore, sessionsStore)(appData, w, r)
+		resp := w.Result()
+
+		assert.Nil(t, err)
+		assert.Equal(t, http.StatusOK, resp.StatusCode)
+		mock.AssertExpectationsForObjects(t, dataStore, sessionsStore, logger, payClient)
+	})
+}
+
+func (m *mockDataStore) withLpaDataInStore() *mockDataStore {
```

Comment:
because of how the mockDataStore works this method name is weird, it is only saying that a `Get` call will not error not that there is any data in the store, as you'd set that like `&mockDataStore{data: myLpaData}`

Reply:
Was struggling with this name - I'll have a play around with some options.